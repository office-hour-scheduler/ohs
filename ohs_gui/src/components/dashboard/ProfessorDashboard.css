.add-course,
.add-section {
  padding-top: 50px;
}
