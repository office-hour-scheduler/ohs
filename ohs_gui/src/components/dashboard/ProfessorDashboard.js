import React from 'react';
import shortid from 'shortid';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import CourseCard from './CourseCard';
import './ProfessorDashboard.css';

class ProfessorDashboard extends React.Component {
  render() {
    const { courses } = this.props;
    console.log(courses);
    const profView = (
      <div id="courses">
        <h1>Current Courses</h1>
        {courses.map(c => (
          <CourseCard course={c} key={shortid.generate()} />
        ))}
        <Link to="/add-course">
          <div className="add-course card-element">
            <span className="fa fa-plus" />
          </div>
        </Link>
      </div>
    );

    return profView;
  }
}

ProfessorDashboard.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  courses: PropTypes.array.isRequired
};

export default ProfessorDashboard;
