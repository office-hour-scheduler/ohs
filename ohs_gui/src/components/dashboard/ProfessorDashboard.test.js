import React from 'react';
import { shallow } from 'enzyme';

import ProfessorDashboard from './ProfessorDashboard';

const COURSES = ['CSC302', 'CSC202'];

// need this to deal with randomly generated ids
jest.mock('shortid', () => {
  return {
    generate: jest.fn(() => 1)
  };
});

it('should render correctly with no props', () => {
  const t = () => {
    shallow(<ProfessorDashboard />);
  };
  expect(t).toThrow(Error);
});

it('test if it renders with params', () => {
  const component = shallow(<ProfessorDashboard courses={COURSES} />);
  expect(component).toMatchSnapshot();
});

it('test if it renders with empty list', () => {
  const component = shallow(<ProfessorDashboard courses={[]} />);
  expect(component).toMatchSnapshot();
});
