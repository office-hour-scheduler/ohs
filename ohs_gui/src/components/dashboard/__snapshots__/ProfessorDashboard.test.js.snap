// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if it renders with empty list 1`] = `
<div
  id="courses"
>
  <h1>
    Current Courses
  </h1>
  <Link
    replace={false}
    to="/add-course"
  >
    <div
      className="add-course card-element"
    >
      <span
        className="fa fa-plus"
      />
    </div>
  </Link>
</div>
`;

exports[`test if it renders with params 1`] = `
<div
  id="courses"
>
  <h1>
    Current Courses
  </h1>
  <CourseCard
    course="CSC302"
    key="1"
  />
  <CourseCard
    course="CSC202"
    key="1"
  />
  <Link
    replace={false}
    to="/add-course"
  >
    <div
      className="add-course card-element"
    >
      <span
        className="fa fa-plus"
      />
    </div>
  </Link>
</div>
`;
